### PR TITLE
[Logs]: Fixed kubernetes ingtegration to tail logs from beginning

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Fixed kubernetes integration that would lead to tail all containers from the end.

### Motivation

Collect all logs from beginning.

### Additional Notes

Kubernetes scanner is likely to change since we plugged the logs-agent to autodiscovery.

### CODE REVIEW

File: pkg/logs/config/integration_config.go [hunk 1]
```diff
@@ -69,6 +69,8 @@ type LogsConfig struct {
 // Validate returns an error if the config is misconfigured
 func (c *LogsConfig) Validate() error {
 	switch {
+	case c.Type == "":
```
Comment (tmichelet) [inline pkg/logs/config/integration_config.go]: when is this happening? Worth a little explanation

File: pkg/logs/input/file/scanner.go [hunk 2]
```diff
@@ -164,7 +164,11 @@ func (s *Scanner) launchTailers(source *config.LogSource) {
 		if _, isTailed := s.tailers[file.Path]; isTailed {
 			continue
 		}
-		s.startNewTailer(file, false)
+		tailFromBeginning := false
+		if source.Config.Identifier != "" {
```
Comment (tmichelet) [inline pkg/logs/input/file/scanner.go]: Same here, worth an explanation

File: pkg/logs/input/kubernetes/scanner.go [hunk 3]
```diff
@@ -136,12 +138,13 @@ func (s *Scanner) getSource(pod *kubelet.Pod, container kubelet.ContainerStatus)
...
 	} else {
 		cfg = &config.LogsConfig{
+			Type: config.FileType,
+			Path: s.getPath(pod, container)
 			Source:  kubernetesIntegration,
 			Service: kubernetesIntegration,
 		}
 	}
-	cfg.Type = config.FileType
-	cfg.Path = s.getPath(pod, container)
```
Comment (tmichelet) [inline pkg/logs/input/kubernetes/scanner.go]: why are we moving these two lines?

Comment (ajacquemot) [reply pkg/logs/input/kubernetes/scanner.go]: Because we need the type to be set for the validation to succeed

**Review by tmichelet — APPROVED** [review]

**Pull request merged**